Sources/Widgets/Representations/ArrowHandleRepresentation/index.js

```diff
@@ -152,6 +173,28 @@ function vtkArrowHandleRepresentation(publicAPI, model) {
       typedArray.points[i * 3 + 1] = coord[1];
       typedArray.points[i * 3 + 2] = coord[2];
 
+      let scale3 = state.getScale3 ? state.getScale3() : [1, 1, 1];
+      scale3 = scale3.map((x) => (x === 0 ? 2 * model.defaultScale : 2 * x));
+
+      const viewMatrix = mat3.create();
+      mat3.fromMat4(viewMatrix, model.viewMatrix);
+      const viewMatrixInv = mat3.create();
+      mat3.invert(viewMatrixInv, viewMatrix);
+
+      const displayOrientation = vec3.create();
+      vec3.transformMat3(displayOrientation, model.orientation, viewMatrixInv);
+      displayOrientation[2] = 0;
+      const displayMatrix = vtkMatrixBuilder
+        .buildFromDegree()
+        .rotateFromDirections([0, 1, 0], displayOrientation)
+        .getMatrix();
+      const displayRotation = mat3.create();
+      mat3.fromMat4(displayRotation, displayMatrix);
+      const rotation = mat3.create();
+      mat3.multiply(rotation, viewMatrix, displayRotation);
+      vec3.transformMat3(scale3, scale3, rotation);
+
+      typedArray.direction.set(rotation, 9 * i);
       typedArray.scale[i] =
         scaleFactor *
         (!state.isVisible || state.isVisible() ? 1 : 0) *
```

Comment:
What is the difference between `state.isVisible` and `handleVisibility` ?

Reply:
state.isVisible refers to the variable from the widget state tree. handleVisibility comes from abstractWidget and is used as a general purpose flag to manage visibility across all widgets